pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt

```diff
@@ -127,25 +126,253 @@ class PushPayloadHelper {
+                        }
+                    }
+                }
+
+                enum class Visibility {
+                    VISIBILITY_UNSPECIFIED,
+                    PRIVATE,
+                    PUBLIC,
+                    SECRET,
+                }
+
+                enum class NotificationPriority {
+                    PRIORITY_UNSPECIFIED,
+                    PRIORITY_MIN,
+                    PRIORITY_LOW,
+                    PRIORITY_DEFAULT,
+                    PRIORITY_HIGH,
+                    PRIORITY_MAX
+                }
+
+                override fun toMap(): Map<String, Any> = buildMap {
+                    title?.let { put("title", it) }
+                    body?.let { put("body", it) }
+                    icon?.let { put("icon", it) }
+                    color?.let { put("color", it) }
+                    sound?.let { put("sound", it) }
+                    tag?.let { put("tag", it) }
+                    bodyLocKey?.let { put("body_loc_key", it) }
+                    bodyLocArgs?.let { put("body_loc_args", it) }
+                    titleLocKey?.let { put("title_loc_key", it) }
+                    titleLocArgs?.let { put("title_loc_args", it) }
+                    channelId?.let { put("channel_id", it) }
+                    ticker?.let { put("ticker", it) }
+                    put("sticky", sticky)
+                    eventTime?.let { put("event_time", it) }
+                    localOnly?.let { put("local_only", it) }
+                    put("notification_priority", notificationPriority.name)
+                    defaultSound?.let { put("default_sound", it) }
+                    defaultVibrateTimings?.let { put("default_vibrate_timings", it) }
+                    defaultLightSettings?.let { put("default_light_settings", it) }
+                    vibrateTimings?.let { put("vibrate_timings", it) }
+                    visibility?.let { put("visibility", it.name) }
+                    notificationCount?.let { put("notification_count", it) }
+                    lightSettings?.let { put("light_settings", it.toMap()) }
+                    image?.let { put("image", it) }
+                }
+            }
+
+            enum class AndroidMessagePriority {
+                NORMAL,
+                HIGH
+            }
+
+            override fun toMap(): Map<String, Any> {
+                return buildMap {
+                    collapseKey?.let { put("collapse_key", it) }
+                    put("priority", priority.name)
+                    ttl?.let { put("ttl", it) }
+                    restrictedPackageName?.let { put("restricted_package_name", it) }
+                    data?.let { put("data", it) }
+                    notification?.let { put("notification", it.toMap()) }
+                    fcmOptions?.let { put("fcm_options", it.toMap()) }
+                    put("direct_boot_ok", directBootOk)
+                }
```

Comment:
Don't we by any chance need to add "click_action" to Notification to handle following situation:

{
  "message": {
    "topic": "news",
    "notification": {
      "title": "Breaking News",
      "body": "New news story available."
    },
    "data": {
      "story_id": "story_12345"
    },
    "android": {
      "notification": {
        "click_action": "TOP_STORY_ACTIVITY",
        "body": "Check out the Top Story"
      }
    },
    "apns": {
      "payload": {
        "aps": {
          "category" : "NEW_MESSAGE_CATEGORY"
        }
      }
    }
  }
}

Reply:
ha, you're right, I missed one! It's in AndroidNotification though. added